File: AtCoder/ACP/BootCamp4b/Medium100_2/cpp/B_Snukes_Coloring_2_ABC_Edit.cpp
#include <bits/stdc++.h>
#define MAX_N 100
using namespace std;

int W, H, N;
int X[MAX_N], Y[MAX_N], A[MAX_N];

void solve() {
  int top = H;
  int bottom = 0;
  int left = 0;
  int right = W;
  for (int i = 0; i < N; ++i) {
    if (A[i] == 1) {
      left = max(left, X[i]);
    } else if (A[i] == 2) {
      right = min(right, X[i]);
    } else if (A[i] == 3) {
      bottom = max(bottom, Y[i]);
    } else {
      top = min(top, Y[i]);
    }
  }
  int res = max(0, top - bottom) * max(0, right - left);
  printf("%d\n", res);
}

int main() {
  cin >> W >> H >> N;
  for (int i = 0; i < N; ++i) {
    cin >> X[i] >> Y[i] >> A[i];
  }
  solve();
}

File: AtCoder/ACP/BootCamp4b/Medium100_2/python/B_Snukes_Coloring_2_ABC_Edit.py
W, H, N = map(int, input().split())
left = 0
right = W
top = H
bottom = 0
for _ in range(N):
    x, y, a = map(int, input().split())
    if a == 1:
        left = max(left, x)
    elif a == 2:
        right = min(right, x)
    elif a == 3:
        bottom = max(bottom, y)
    else:
        top = min(top, y)
res = max(0, top - bottom) * max(0, right - left)
print(res)
